libraries/fc/src/log/logger_config.cpp

```diff
@@ -84,4 +84,15 @@ namespace fc {
       cfg.loggers.push_back( dlc );
       return cfg;
    }
+
+   static thread_local std::string thread_name;
+   void set_thread_name( const string& name ) {
+      thread_name = name;
+   }
+   const string& get_thread_name() {
+      static int thread_count = 0;
+      if( thread_name == string() )
```

Comment:
While I'm sure the compiler will handle this well enough, `thread_name.empty()` is a more canonical way to perform this check and it won't allocate any temporaries in un-optimized builds.